### PR TITLE
SEQNG-918 Fixed several bugs in the TCS configuration.

### PR DESCRIPTION
**Pull request by jluhrs**

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala [hunk 1]
```diff
@@ -104,11 +104,11 @@ final case class Tcs private (tcsController: TcsController,
           ),
           tag[P2Config](calcGuiderConfig(
             calcGuiderInUse(c.tcsGuide, TipTiltSource.PWFS2, M1Source.PWFS2),
-            config.guideWithP1)
+            config.guideWithP2)
           ),
           tag[OIConfig](calcGuiderConfig(
             calcGuiderInUse(c.tcsGuide, TipTiltSource.OIWFS, M1Source.OIWFS) | aoUsesOI,
-            config.guideWithP1)
+            config.guideWithOI)
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala]: The evils of copy and paste

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala [hunk 2]
```diff
@@ -167,9 +167,9 @@ object Tcs {
     val gwp2 = config.extractAs[StandardGuideOptions.Value](TELESCOPE_KEY / GUIDE_WITH_PWFS2_PROP).toOption
     val gwoi = config.extractAs[StandardGuideOptions.Value](TELESCOPE_KEY / GUIDE_WITH_OIWFS_PROP).toOption
     val gwao = config.extractAs[StandardGuideOptions.Value](TELESCOPE_KEY / GUIDE_WITH_AOWFS_PROP).toOption
-    val offsetp = config.extractAs[Double](TELESCOPE_KEY / P_OFFSET_PROP).toOption
+    val offsetp = config.extractAs[String](TELESCOPE_KEY / P_OFFSET_PROP).toOption.flatMap(_.parseDoubleOption)
       .map(Arcseconds(_):Angle).map(tag[OffsetP](_))
-    val offsetq = config.extractAs[Double](TELESCOPE_KEY / Q_OFFSET_PROP).toOption
+    val offsetq = config.extractAs[String](TELESCOPE_KEY / Q_OFFSET_PROP).toOption.flatMap(_.parseDoubleOption)
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala]: Shouldn't the offsets be sent as Double ?

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala [hunk 3]
```diff
@@ -209,16 +209,32 @@ object TcsControllerEpics extends TcsController {
         setOiwfs, EpicsTcsConfig.oiwfs ^|-> GuiderConfig.detector)
     ).collect{ case Some(x) => x }
 
+
+  def tagIso[B, T]: Iso[B@@T, B] = Iso.apply[B@@T, B](x => x)(tag[T](_))
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala]: Nice!

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsController.scala [hunk 4]
```diff
@@ -412,9 +412,10 @@ object TcsController {
     // Gemini Adaptive Optics System (GeMS or Altair)
     case object Gaos   extends Subsystem
 
-    implicit val order: Order[Subsystem] = Order.allEqual
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsController.scala]: What was I thinking? Of course this broke all Set builds and `contains` tests.

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala [hunk 5]
```diff
@@ -149,10 +149,10 @@ object TcsControllerEpics extends TcsController {
...
                            tcs: TcsConfig): SeqAction[Unit] = {
     def configParams(current: EpicsTcsConfig): List[EpicsTcsConfig => SeqAction[EpicsTcsConfig]] = List(
-      applyParam(subsystems.contains(Subsystem.PWFS1), current.pwfs1.tracking, tcs.gds.oiwfs.tracking,
-        setProbeTrackingConfig(TcsEpics.instance.oiwfsProbeGuideCmd), EpicsTcsConfig.pwfs1 ^|-> GuiderConfig.tracking),
-      applyParam(subsystems.contains(Subsystem.PWFS2), current.pwfs2.tracking, tcs.gds.oiwfs.tracking,
-        setProbeTrackingConfig(TcsEpics.instance.oiwfsProbeGuideCmd), EpicsTcsConfig.pwfs2 ^|-> GuiderConfig.tracking),
+      applyParam(subsystems.contains(Subsystem.PWFS1), current.pwfs1.tracking, tcs.gds.pwfs1.tracking,
+        setProbeTrackingConfig(TcsEpics.instance.pwfs1ProbeGuideCmd), EpicsTcsConfig.pwfs1 ^|-> GuiderConfig.tracking),
+      applyParam(subsystems.contains(Subsystem.PWFS2), current.pwfs2.tracking, tcs.gds.pwfs2.tracking,
+        setProbeTrackingConfig(TcsEpics.instance.pwfs2ProbeGuideCmd), EpicsTcsConfig.pwfs2 ^|-> GuiderConfig.tracking),
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala]: More copy and paste mistakes.

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala (hunk 2 above)
Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala]: They should!

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala [hunk 6]
```diff
@@ -209,16 +209,32 @@ object TcsControllerEpics extends TcsController {
         setOiwfs, EpicsTcsConfig.oiwfs ^|-> GuiderConfig.detector)
     ).collect{ case Some(x) => x }
 
+
+  def tagIso[B, T]: Iso[B@@T, B] = Iso.apply[B@@T, B](x => x)(tag[T](_))
+
   def calcGuideOff(current: EpicsTcsConfig, demand: TcsConfig): TcsConfig = {
-    if(willMove(current, demand)) (TcsConfig.gds.modify(
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala]: The main change here is that in the guide-off step, I should only turn thing off. Before I was returning the original demand, which may have some things turned on

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala (hunk 3 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsControllerEpics.scala]: To be used cautiously, with the risk of erasing the benefits of `tag`

**Pull request merged**